### ISSUE TITLE
DecoderErrorDeserialiseFailure after db-sync-restart

### ISSUE THREAD

**Issue by PegasusPool** [input-output-hk/cardano-db-sync#506]
```
cardano-db-sync-extended version
cardano-db-sync-extended 7.1.0 - linux-x86_64 - ghc-8.6
git revision 196ba716c425a0b7c75741c168f6a6d7edaee1fc
```

cardano-db-sync has been running pretty much without issues for the last few months.

I recently restarted the service and I cannot start it up again. I'd really like to avoid a full resync so if the following error could be resolved some other way that'd be great:

```
Started Cardano-db-sync.
[db-sync-node:Info:4] [2021-01-25 06:26:08.88 UTC] NetworkMagic: 764824073
[db-sync-node:Info:4] [2021-01-25 06:26:08.92 UTC] Initial genesis distribution present and correct
[db-sync-node:Info:4] [2021-01-25 06:26:08.92 UTC] Total genesis supply of Ada: 31112484745.000000
[db-sync-node:Info:4] [2021-01-25 06:26:08.93 UTC] Inserting Shelley Genesis distribution
[db-sync-node:Info:4] [2021-01-25 06:26:08.94 UTC] epochPluginOnStartup: Checking
[db-sync-node:Info:4] [2021-01-25 06:26:08.94 UTC] localInitiatorNetworkApplication: connecting to node via "/opt/cardano/cnode/sockets/node.socket"
[db-sync-node.Subscription:Notice:30] [2021-01-25 06:26:08.94 UTC] [String "Starting Subscription Worker, valency 1",String "SubscriptionTrace"]
[db-sync-node.Subscription:Notice:31] [2021-01-25 06:26:08.94 UTC] [String "Connection Attempt Start, destination LocalAddress \"/opt/cardano/cnode/sockets/node.socket\"",String "SubscriptionTrace"]
[db-sync-node.Subscription:Notice:31] [2021-01-25 06:26:08.94 UTC] [String "Connection Attempt End, destination LocalAddress \"/opt/cardano/cnode/sockets/node.socket\" outcome: ConnectSuccessLast",String "SubscriptionTrace"]
[db-sync-node.Handshake:Info:31] [2021-01-25 06:26:08.94 UTC] [String "Send (ClientAgency TokPropose,MsgProposeVersions (fromList [(NodeToClientV_1,TInt 764824073),(NodeToClientV_2,TInt 764824073),(NodeToClientV_3,TInt 764824073),(NodeToClientV_4,TInt 764824073),(NodeToClientV_5,TInt 764824073),(NodeToClientV_6,TInt 764824073)]))",String "LocalHandshakeTrace",String "ConnectionId {localAddress = LocalAddress \"\", remoteAddress = LocalAddress \"/opt/cardano/cnode/sockets/node.socket\"}"]
[db-sync-node.Handshake:Info:31] [2021-01-25 06:26:08.94 UTC] [String "Recv (ServerAgency TokConfirm,MsgAcceptVersion NodeToClientV_6 (TInt 764824073))",String "LocalHandshakeTrace",String "ConnectionId {localAddress = LocalAddress \"\", remoteAddress = LocalAddress \"/opt/cardano/cnode/sockets/node.socket\"}"]
[db-sync-node.Mux:Info:31] [2021-01-25 06:26:08.94 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: Handshake Client end, duration 0.000702567s
[db-sync-node.Mux:Info:32] [2021-01-25 06:26:08.94 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: State: Mature
[db-sync-node:Info:36] [2021-01-25 06:26:08.94 UTC] Starting chainSyncClient
[db-sync-node.Subscription:Notice:30] [2021-01-25 06:26:08.97 UTC] [String "Required subscriptions started",String "SubscriptionTrace"]
[db-sync-node:Info:36] [2021-01-25 06:26:08.97 UTC] Database tip is at slot 19988991, block 5254425
[db-sync-node:Info:41] [2021-01-25 06:26:08.97 UTC] Running DB thread
[db-sync-node:Info:41] [2021-01-25 06:26:08.99 UTC] Rolling back to slot 19988991, hash d52012289215f2f5f206461ba9976a4d54eafe35f26d363ab00e76758c173d51
[db-sync-node:Info:41] [2021-01-25 06:26:08.99 UTC] Deleting slots numbered: []
[db-sync-node:Info:44] [2021-01-25 06:26:08.99 UTC] getHistoryInterpreter: acquired
[db-sync-node:Error:41] [2021-01-25 06:26:10.16 UTC] runDBThread: FatalError {fatalErrorMessage = "DecoderErrorDeserialiseFailure \"Ledger state file\" (DeserialiseFailure 92181400 \"end of input\")"}
[db-sync-node:Error:36] [2021-01-25 06:26:10.16 UTC] ChainSyncWithBlocksPtcl: FatalError {fatalErrorMessage = "DecoderErrorDeserialiseFailure \"Ledger state file\" (DeserialiseFailure 92181400 \"end of input\")"}
[db-sync-node.Mux:Info:32] [2021-01-25 06:26:10.16 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: State: Dead
[db-sync-node.Mux:Notice:32] [2021-01-25 06:26:10.16 UTC] Bearer on ConnectionId {localAddress = LocalAddress "", remoteAddress = LocalAddress "/opt/cardano/cnode/sockets/node.socket"} event: Miniprotocol MiniProtocolNum 5 InitiatorDir terminated with exception FatalError {fatalErrorMessage = "DecoderErrorDeserialiseFailure \"Ledger state file\" (DeserialiseFailure 92181400 \"end of input\")"}
[db-sync-node.Subscription:Error:31] [2021-01-25 06:26:10.16 UTC] [String "Application Exception: LocalAddress \"/opt/cardano/cnode/sockets/node.socket\" FatalError {fatalErrorMessage = \"DecoderErrorDeserialiseFailure \\\"Ledger state file\\\" (DeserialiseFailure 92181400 \\\"end of input\\\")\"}",String "SubscriptionTrace"]
cardano-db-sync-extended: FatalError {fatalErrorMessage = "DecoderErrorDeserialiseFailure \"Ledger state file\" (DeserialiseFailure 92181400 \"end of input\")"}
[db-sync-node.ErrorPolicy:Error:4] [2021-01-25 06:26:10.16 UTC] [String "ErrorPolicyLocalNodeError (ApplicationExceptionTrace (FatalError {fatalErrorMessage = \"DecoderErrorDeserialiseFailure \\\"Ledger state file\\\" (DeserialiseFailure 92181400 \\\"end of input\\\")\"}))",String "ErrorPolicyTrace",String "LocalAddress \"/opt/cardano/cnode/sockets/node.socket\""]
cardano-db-sync.service: Main process exited, code=exited, status=1/FAILURE
cardano-db-sync.service: Failed with result 'exit-code'.
```

**Comment by PegasusPool** [comment]
So it looks like somehow the ledger state was corrupted due to the restart (as suggested by the error message).

I'm leaving this here for the record as I seem to have resolved this issue by doing the followings:

- Inspected the contents of directory `db-sync-extended/state`
```
19988877.lstate  19988879.lstate  19988880.lstate  19988908.lstate  19988933.lstate  19988942.lstate  19988949.lstate  19988955.lstate  19988991.lstate
```
- Removed all but the oldest lstate file which was not too far behind the tip
- Started cardano-db-sync which rolled back to the previous (valid state)
```
Rolling back to slot 19988877, hash bde6e25d35699d0d59ff95675b30f0569606e049de1fa423cfd603bf5bf2dd60
Deleting slots numbered: [19988991,19988955,19988949,19988942,19988933,19988908,19988880,19988879]
``` 
- cardano-db-sync caught up with the tip and has been syncing fine.

I'll close the issue but let me know if you need any more info.

**Comment by erikd** [comment]
There is a specific fix for this in 0db9898a12542b4e9ba41d4266840acd55b012e3 .
